pkg/parse/parse.go

```diff
@@ -589,11 +590,30 @@ func SystemContextFromOptions(c *cobra.Command) (*types.SystemContext, error) {
 		ctx.RegistriesDirPath = regConfDir
 	}
 	ctx.DockerRegistryUserAgent = fmt.Sprintf("Buildah/%s", buildah.Version)
-	if os, err := c.Flags().GetString("override-os"); err == nil {
+
+	if os, err := c.Flags().GetString("override-os"); err == nil && os != runtime.GOOS {
+		ctx.OSChoice = os
+	} else if os, err := c.Flags().GetString("os"); err == nil && os != runtime.GOOS {
 		ctx.OSChoice = os
+	} else {
+		ctx.OSChoice = runtime.GOOS
+	}
+
+	if arch, err := c.Flags().GetString("override-arch"); err == nil && arch != runtime.GOARCH {
+		ctx.ArchitectureChoice = arch
+	} else if arch, err := c.Flags().GetString("arch"); err == nil && arch != runtime.GOARCH {
+		ctx.ArchitectureChoice = arch
+	} else {
+		ctx.ArchitectureChoice = runtime.GOARCH
 	}
-	if arch, err := c.Flags().GetString("override-arch"); err == nil {
+
+	if pf, err := c.Flags().GetString("platform"); err == nil && pf != DefaultPlatform() {
+		os, arch, err := parsePlatform(pf)
+		if err != nil {
+			return nil, errors.Wrapf(err, "unable to parse platform %s", pf)
+		}
 		ctx.ArchitectureChoice = arch
+		ctx.OSChoice = os
 	}
```

Comment:
This part is a bit tricky because we have some sort of implicit priority, but I think it should not harm the overall behavior.

Reply:
We need new fields for this in the executer (see the other comments).

Reply:
Yep, added the values to the executor.